beamer/agent/middleware.py

```diff
@@ -282,3 +312,34 @@ def rate_limiter(
 ) -> Callable[[RPCEndpoint, Any], RPCResponse]:
     state = _RateLimiterState()
     return functools.partial(_rate_limiter, make_request=make_request, w3=w3, state=state)
+
+
+def max_fee_setter(
+    make_request: _MakeRequest, _w3: Web3, chain_id: ChainId
+) -> Callable[[RPCEndpoint, Any], RPCResponse]:
+    def middleware(method: RPCEndpoint, params: Any) -> RPCResponse:
+        if method != "eth_sendTransaction":
+            return make_request(method, params)
+        priority_fee_response = make_request(RPCEndpoint("eth_maxPriorityFeePerGas"), [])
+        if _result_ok(priority_fee_response):
+            priority_fee = priority_fee_response["result"]
+        else:
+            return priority_fee_response
+        latest_block_number = _BLOCK_STORAGE[chain_id].get_latest_block_number()
```

Comment:
reading this part now, I would actually give the `_BlockCache` class a bit more power to directly get the latest block result 

like 

```py
@property
def latest_block() RPCResponse | None:
       return self.get_block(self._latest_key))
```

Reply:
This way we could save some boilerplate in hte max fee setter middleware